lms/views/dashboard/api/course.py

```diff
@@ -49,10 +52,15 @@ def __init__(self, request) -> None:
         schema=ListCoursesSchema,
     )
     def courses(self) -> APICourses:
+        filter_by_h_userids = self.request.parsed_params.get("h_userids")
+        filter_by_assignment_ids = self.request.parsed_params.get("assignment_ids")
+
         courses = self.course_service.get_courses(
             instructor_h_userid=self.request.user.h_userid
             if self.request.user
             else None,
+            h_userids=filter_by_h_userids,
```

Comment:
These already existed on the service to filter the courses for the metrics endpoints, so this PR just needs to pass the parameters.